analyzers/src/SonarAnalyzer.CFG/ShimLayer/TypeInfoExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
```

Comment:
I kept the license header from StyleCop analyzers and all classes/structs/enums needed are in one file because I wasn't sure what the right license header/file location is for our own shims.

I think we should also create a PR in the style cop repo once we merged this PR.

Reply:
Create an issue on StyleCop repo to ask if they want it. And then feel free to create the PR on their side too.

Reply:
Their license in this directory is fine

Reply:
> Create an issue on StyleCop repo to ask if they want it. And then feel free to create the PR on their side too.

DotNetAnalyzers/StyleCopAnalyzers#3560